### PR TITLE
Added content from blog post to troubleshooting page

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: content/docs/next-release/troubleshooting.md [hunk 1]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+## Sampling strategy
+
+Before everything else, make sure to confirm what is the sampling strategy. By default, Jaeger uses a probabilistic sampling strategy, with a 1/1000th chance that the span will be reported.
+
+Typically, the sampling strategy can be set via the environment variables `JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM`, but refer to the Jaeger Client’s documentation for the language you are using for more details about which sampling strategies are available. When using the Jaeger Java Client, the strategy is usually printed out via the logging facility provided by the instrumented application when creating the tracer:
+
+```
+2018-12-10 16:41:25 INFO  Configuration:236 - Initialized  tracer=JaegerTracer(..., sampler=ConstSampler(decision=true,  tags={sampler.type=const, sampler.param=true}), ...)
+```
+
```
Comment (objectiser) [inline content/docs/next-release/troubleshooting.md]: Would it be worth clearing stating that, to diagnose why traces may not be received, it is recommended to configure the `const` sampler type with param `1`?

File: content/docs/next-release/troubleshooting.md [hunk 2]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+## Logging reporter
+
+Some Jaeger clients are able to log the spans that are being reported to the logging facility provided by the instrumented application. Typically, this can be done by setting the environment variable `JAEGER_REPORTER_LOG_SPANS` to `true`, but refer to the Jaeger Client’s documentation for the language you are using. When using the Jaeger _Java_ Client, spans are reported like the following:
+
+    2018-12-10 17:20:54 INFO  LoggingReporter:43 - Span reported:  e66dc77b8a1e813b:6b39b9c18f8ef082:a56f41e38ca449a4:1 -  getAccountFromCache
+
+The log entry above contains three IDs: the trace ID `e66dc77b8a1e813b`, the span’s ID `6b39b9c18f8ef082` and the span’s parent ID `a56f41e38ca449a4`. When the backend components have the log level set to `debug`, the span and trace IDs should be visible on their standard output (see more about that below, under “Increase the logging in the backend components”).
+
+The logging reporter follows the sampling decision made by the sampler, meaning that if the span is logged, it should also reach the the agent or collector.
```
Comment (objectiser) [inline content/docs/next-release/troubleshooting.md]: "the the"

File: content/docs/next-release/troubleshooting.md [hunk 3]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+
+## Bypass the Jaeger Agent
+
+By default, the Jaeger Agent receives spans via UDP from the Jaeger Client. As some networking setups might drop or block UDP packets, the Jaeger client can be configured to bypass the agent, sending spans directly to the collector. Some clients, such as the Jaeger Java Client, support the environment variable `JAEGER_ENDPOINT` which can be used to specify the collector’s location, such as http://jaeger-collector:14268/api/traces. Refer to the Jaeger Client’s documentation for the language you are using. When setting this environment variable, the Jaeger Java Client logs the following when the tracer is created:
```
Comment (objectiser) [inline content/docs/next-release/troubleshooting.md]: Should this say "on localhost" - a typical problem is that the agent may not actually be running locally (e.g. when daemonset) - so might be worth highlighting that situation first?

File: content/docs/next-release/troubleshooting.md (hunk 1 above)
Comment (yurishkuro) [reply content/docs/next-release/troubleshooting.md]: +1

File: content/docs/next-release/troubleshooting.md (hunk 2 above)
Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: Good catch! It was a problem in the original blog post as well.

File: content/docs/next-release/troubleshooting.md (hunk 3 above)
Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: Should I say instead that the client *sends* to `localhost` by default? The agent won't care of the spans came from localhost or not.

Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: Like this:

> By default, the Jaeger Client is configured to send spans via UDP to a Jaeger Agent running on `localhost`.

Comment (objectiser) [reply content/docs/next-release/troubleshooting.md]: +1 sounds good.

File: content/docs/next-release/troubleshooting.md [hunk 4]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+    curl http://jaeger-collector:14268/metrics
+
+The following metrics are of special interest:
+
+    jaeger_collector_spans_received
+    jaeger_collector_spans_saved_by_svc
+    jaeger_collector_traces_received
+    jaeger_collector_traces_saved_by_svc
+
+The first two metrics should have similar values for the same service. Similarly, the two `traces` metrics should also have similar values. For instance, this is an example of a sane setup:
```
Comment (JStickler) [inline content/docs/next-release/troubleshooting.md]: Sane?  

Is the example a healthy configuration?  A sensible configuration?  I'm not sure what you were trying to get across here.

File: content/docs/next-release/troubleshooting.md [hunk 5]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
 weight: 10
 ---
 
-We are preparing a troubleshooting guide for you. In the meantime, make sure to check the blog post ["Help! Something is wrong with my Jaeger installation!"](https://medium.com/jaegertracing/help-something-is-wrong-with-my-jaeger-installation-68874395a7a6)
\ No newline at end of file
+Jaeger is composed of different components, each potentially running in its own host. It might be the case that one of these moving parts isn't working properly, causing spans to not be processed and stored. When something goes wrong, make sure to check the items listed here.
+
+## Sampling strategy
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: "Verify the sampling strategy" sounds good

File: content/docs/next-release/troubleshooting.md [hunk 6]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+
+Before everything else, make sure to confirm what is the sampling strategy. By default, Jaeger uses a probabilistic sampling strategy, with a 1/1000th chance that the span will be reported. For development purposes or for low-traffic scenarios, however, it's appropriate to sample every trace.
+
+Typically, the sampling strategy can be set via the environment variables `JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM`, but refer to the Jaeger Client's documentation for the language you are using for more details about which sampling strategies are available. When using the Jaeger Java Client, the strategy is usually printed out via the logging facility provided by the instrumented application when creating the tracer:
+
+    2018-12-10 16:41:25 INFO  Configuration:236 - Initialized  tracer=JaegerTracer(..., sampler=ConstSampler(decision=true,  tags={sampler.type=const, sampler.param=true}), ...)
+
+To configure the client to sample every trace, set the `JAEGER_SAMPLER_TYPE` environment variable to `const` and set `JAEGER_SAMPLER_PARAM` to `true`
+
+## Logging reporter
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: How about "Use the logging reporter" ?

File: content/docs/next-release/troubleshooting.md (hunk 4 above)
Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: "of a setup that is working appropriately" ?

Comment (jpkrohling) [reply content/docs/next-release/troubleshooting.md]: Or rather, "working as expected".

**Comment by jpkrohling** [comment]
I just pushed some changes based on the review comments. Sorry for taking so long to work on them, I didn't have it on my radar :-/

File: content/docs/next-release/troubleshooting.md [hunk 7]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+    {"level":"debug","ts":1544458854.5406284,"caller":"app/span_handler.go:90","msg":"Span batch processed by the collector.","ok":true}
+    {"level":"debug","ts":1544458854.5406587,"caller":"app/span_processor.go:105","msg":"Span written to the storage by the collector","trace-id":"e66dc77b8a1e813b","span-id":"6b39b9c18f8ef082"}
+    {"level":"debug","ts":1544458854.54068,"caller":"app/span_processor.go:105","msg":"Span written to the storage by the collector","trace-id":"e66dc77b8a1e813b","span-id":"d92976b6055e6779"}
+    {"level":"debug","ts":1544458854.5406942,"caller":"app/span_processor.go:105","msg":"Span written to the storage by the collector","trace-id":"e66dc77b8a1e813b","span-id":"a56f41e38ca449a4"}
+
+## Check the /metrics endpoint
+
+For the cases where it's not possible or desirable to increase the logging in the collector's side, the `/metrics` endpoint can be used to check if spans for specific services were received. Assuming that the Jaeger Collector is available under a host named `jaeger-collector`, here's a sample `curl` call to obtain the metrics:
+
+    curl http://jaeger-collector:14268/metrics
```
Comment (objectiser) [inline content/docs/next-release/troubleshooting.md]: This should now be port 14269.

File: content/docs/1.13/troubleshooting.md [hunk 8]
```diff
@@ -4,4 +4,68 @@ description: Solve commonly encountered issues
...
+
+## Verify the sampling strategy
+
+Before everything else, make sure to confirm what is the sampling strategy. By default, Jaeger uses a probabilistic sampling strategy, with a 1/1000th chance that the span will be reported. For development purposes or for low-traffic scenarios, however, it's appropriate to sample every trace.
+
+Typically, the sampling strategy can be set via the environment variables `JAEGER_SAMPLER_TYPE` and `JAEGER_SAMPLER_PARAM`, but refer to the Jaeger Client's documentation for the language you are using for more details about which sampling strategies are available. When using the Jaeger Java Client, the strategy is usually printed out via the logging facility provided by the instrumented application when creating the tracer:
+
+    2018-12-10 16:41:25 INFO  Configuration:236 - Initialized  tracer=JaegerTracer(..., sampler=ConstSampler(decision=true,  tags={sampler.type=const, sampler.param=true}), ...)
+
+When diagnosing why spans are not being received by other components, make sure to configure the client to sample every trace, setting the `JAEGER_SAMPLER_TYPE` environment variable to `const` and the `JAEGER_SAMPLER_PARAM` to `true`.
```
Comment (yurishkuro) [inline content/docs/1.13/troubleshooting.md]: > `JAEGER_SAMPLER_PARAM` to `true`

Is that correct? The PARAM is usually numeric.

Comment (yurishkuro) [reply content/docs/1.13/troubleshooting.md]: https://github.com/jaegertracing/documentation/pull/274

**Pull request merged**